oepcie/yuanphil_impl/oe_ctx.h

```diff
@@ -0,0 +1,31 @@
+#ifndef _INCL_GUARD_ONE
+#define _INCL_GUARD_ONE
+
+#include "oe_ctx_impl.h"
+
+typedef struct oe_ctx_impl *oe_ctx;
```

Comment:
So within oe_ctx.c, you can define oe_ctx_impl and freely use it. The API user (who just includes this header) will only know about this declaration which provides no information about the internals of oe_ctx_impl. You don't need to use a separate set of source and header files to achieve this.